packages/flutter/lib/src/gestures/converter.dart

```diff
@@ -283,6 +283,9 @@ abstract final class PointerEventConverter {
                 position: position,
                 scrollDelta: scrollDelta,
                 embedderId: datum.embedderId,
+                whenIgnored: () {
+                  packet.handled = false;
```

Comment:
> What happens when after this was set to false somebody else goes in and sets handled to true again?

The only good thing about the API I have now is that it's very simple, but I agree it's weak. I just used this to not have to potentially rewrite a bunch of extra code just to pass a boolean back :)

Yegor has a cool suggestion in the engine side PR, where we can instead return a `DataPacketResponse` object from the Framework (see [this comment](https://github.com/flutter/engine/pull/51566#discussion_r1532961725)).

> Is doing this per PointerDataPacket the right granularity? I believe, in theory a PointerDataPacket could contain information about multiple unrelated pointers, some of which may be handled by the framework while others need to be rejected.

Good question. For "signals" it is probably good enough. For more complex gestures, I'm not sure. On the web, you either `preventDefault` of a browser event immediately, or it will do its default thing, so there's not much room to prevent a "buffered" event, or anything that has been synthetically created because the framework needs to see it.